spec/notifiers/blue_state_notifier_spec.rb

```diff
@@ -27,4 +27,22 @@
     end
     BlueStateNotifier.new.process(charge)
   end
+
+  it "should push all tags as sources into BSD" do
+    tag_names = ['takecharge']
+    (0...5).each do
+      charge.tags << (tag = create(:tag, organization: organization))
+      tag_names << "takecharge:#{tag.name}"
+    end
+    charge.save!
+
+    expect(BlueStateDigital::Contribution).to receive(:new) do |options|
+      sources = options[:source]
+      expect(sources.length).to eq tag_names.length
+      tag_names.each { |tag_name| expect(sources).to include(tag_name) }
```

Comment:
you might be able to write this as <code>expect(sources).to contain_exactly(tag_names)</code>